From f3b07985b09cdea3a37fdd7cd826ae0a05486816 Mon Sep 17 00:00:00 2001
From: Dongwon Kim <dongwon.kim@intel.com>
Date: Thu, 4 Sep 2025 12:55:29 -0700
Subject: [PATCH 2/3] hw/display/virtio-gpu: Properly free current_cursor

s->current_cursor should be freed properly instead of
brutally freeing the structure without considering ref
count.

Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>
---
 hw/display/virtio-gpu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/display/virtio-gpu.c b/hw/display/virtio-gpu.c
index c90e242e12..11f33daed5 100644
--- a/hw/display/virtio-gpu.c
+++ b/hw/display/virtio-gpu.c
@@ -64,7 +64,7 @@ void virtio_gpu_update_cursor_data(VirtIOGPU *g,
              * The assumption is the cursor image is square.
              */
             int width = sqrt(res->blob_size / 4);
-            g_free(s->current_cursor);
+            cursor_unref(s->current_cursor);
             s->current_cursor = cursor_alloc(width, width);
         }
         data = res->blob;
-- 
2.34.1

